Build failure on illumos (omnios-r151032-19f7bd2ae5)

Please read the [instruction](https://github.com/v2ray/v2ray-core/blob/master/.github/SUPPORT.md) and answer the following questions before submitting your issue. Thank you.
Please answer all the questions with enough information. All issues not following this template will be closed immediately.
If you are not sure if your question is truely a bug in V2Ray, please discuss it [here](https://github.com/v2ray/discussion/issues) first.

I'm sure it's not a "bug" for which this template is intended, but I can't find a more proper one.

1) What version of V2Ray are you using (If you deploy different version on server and client, please explicitly point out)?

master branch in this repo

2) What's your scenario of using V2Ray? E.g., Watching YouTube videos in Chrome via Socks/VMess proxy.

I don't think it's relevant.

3) What did you see? (Please describe in detail, such as timeout, fake TLS certificate etc)

```
$ go test
# v2ray.com/core/common/buf
common/buf/readv_posix.go:22:4: cannot use &b.v[0] (type *byte) as type *int8 in field value
common/buf/readv_posix.go:30:29: undefined: syscall.SYS_READV
FAIL	v2ray.com/core [build failed]
```

4) What's your expectation?

build complete successfully

5) Please attach your configuration file (**Mask IP addresses before submit this issue**).

Server configuration:

```javascript
    // Please attach your server configuration here.
```

Client configuration:

```javascript
    // Please attach your client configuration here.
```

6) Please attach error logs, especially the bottom lines if the file is large. Error log file is usually at `/var/log/v2ray/error.log` on Linux.

Server error log:

```javascript
    // Please attach your server error log here.
```

Client error log:

```javascript
    // Please attach your client error log here.
```

7) Please attach access log. Access log is usually at '/var/log/v2ray/access.log' on Linux.

```javascript
    // Please attach your server access log here.
```

8) Other configurations (such as Nginx) and logs.

9) If V2Ray doesn't run, please attach output from `--test`.

The command is usually `/usr/bin/v2ray/v2ray --test --config /etc/v2ray/config.json`, but may vary according to your scenario.

10) If V2Ray service doesn't run, please attach journal log.

Usual command is `journalctl -u v2ray`.

Please review your issue before submitting.

Sorry for choosing an improper template, but I can't find a more proper one, and I have to choose one. I can bring this up in discussion but it will be hard to cite the issue in commit message, and I've already got a PR prepared.

The `syscall` package is missing lots of definitions on Solaris/illumos, which makes it basically unusable for us on those platforms. And unfortunately, `syscall` package is already freezed, which means its unlikely that those missing definitions will be added in the future.

Therefore, I suggest switching to `x/sys/unix` whenever possible, which
+ Is active
+ Basically the same with `syscall` (if not better)

I've updated multiReader implementation on illumos to use x/sys/unix, and it doesn't take lots of efforts, so I think it's worth considering?

More details in the PR.

--- a/common/buf/readv_posix.go
+++ b/common/buf/readv_posix.go
@@ -1,5 +1,6 @@
 // +build !windows
 // +build !wasm
+// +build !illumos
 
 package buf
 

--- a/common/buf/readv_unix.go
+++ b/common/buf/readv_unix.go
@@ -1,0 +1,36 @@
+// +build illumos
+
+package buf
+
+import "golang.org/x/sys/unix"
+
+type unixReader struct {
+	iovs [][]byte
+}
+
+func (r *unixReader) Init(bs []*Buffer) {
+	iovs := r.iovs
+	if iovs == nil {
+		iovs = make([][]byte, 0, len(bs))
+	}
+	for _, b := range bs {
+		iovs = append(iovs, b.v)
+	}
+	r.iovs = iovs
+}
+
+func (r *unixReader) Read(fd uintptr) int32 {
+	n, e := unix.Readv(int(fd), r.iovs)
+	if e != nil {
+		return -1
+	}
+	return int32(n)
+}
+
+func (r *unixReader) Clear() {
+	r.iovs = r.iovs[:0]
+}
+
+func newMultiReader() multiReader {
+	return &unixReader{}
+}

--- a/transport/internet/domainsocket/listener.go
+++ b/transport/internet/domainsocket/listener.go
@@ -9,8 +9,9 @@ import (
 	gotls "crypto/tls"
 	"os"
 	"strings"
-	"syscall"
 
+	"golang.org/x/sys/unix"
+	
 	"v2ray.com/core/common"
 	"v2ray.com/core/common/net"
 	"v2ray.com/core/transport/internet"
@@ -104,7 +105,7 @@ func (fl *fileLocker) Acquire() error {
 	if err != nil {
 		return err
 	}
-	if err := syscall.Flock(int(f.Fd()), syscall.LOCK_EX); err != nil {
+	if err := unix.Flock(int(f.Fd()), unix.LOCK_EX); err != nil {
 		f.Close()
 		return newError("failed to lock file: ", fl.path).Base(err)
 	}
@@ -113,7 +114,7 @@ func (fl *fileLocker) Acquire() error {
 }
 
 func (fl *fileLocker) Release() {
-	if err := syscall.Flock(int(fl.file.Fd()), syscall.LOCK_UN); err != nil {
+	if err := unix.Flock(int(fl.file.Fd()), unix.LOCK_UN); err != nil {
 		newError("failed to unlock file: ", fl.path).Base(err).WriteToLog()
 	}
 	if err := fl.file.Close(); err != nil {

--- a/transport/internet/sockopt_other.go
+++ b/transport/internet/sockopt_other.go
@@ -1,4 +1,4 @@
-// +build js dragonfly netbsd openbsd
+// +build js dragonfly netbsd openbsd solaris
 
 package internet
 

